plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,243 @@
+    type: str
+    required: false
+requirements:
+- PyGithub>=1.54
+notes:
+- For python3, PyGithub>=1.54 should be used.
+- "For python3.5, PyGithub==1.54 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-54-november-30-2020."
+- "For python2.7, PyGithub==1.45 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-45-december-29-2019."
+author:
+- Álvaro Torres Cogollo (@atorrescogollo)
+'''
+
+EXAMPLES = '''
+- name: Create a Github Repo
+  community.general.github_repo:
+    access_token: mytoken
+    organization: MyOrganization
+    name: myrepo
+    description: "Just for fun"
+    private: yes
+    state: present
+  register: result
+- name: Delete the repo
+  community.general.github_repo:
+    username: octocat
+    password: password
+    organization: MyOrganization
+    name: myrepo
+    state: absent
+  register: result
+'''
+
+RETURN = '''
+repo:
+  description: Repository information as JSON. See U(https://docs.github.com/en/rest/reference/repos#get-a-repository).
+  returned: success and I(state) is C(present)
+  type: dict
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import sys
+
+GITHUB_IMP_ERR = None
+try:
+    from github import Github, GithubException
+    from github.GithubException import UnknownObjectException
+    HAS_GITHUB_PACKAGE = True
+except Exception:
+    GITHUB_IMP_ERR = traceback.format_exc()
+    HAS_GITHUB_PACKAGE = False
+
+
+def authenticate(username=None, password=None, access_token=None):
+    if access_token:
+        return Github(base_url="https://api.github.com:443", login_or_token=access_token)
+    else:
+        return Github(base_url="https://api.github.com:443", login_or_token=username, password=password)
+
+
+def create_repo(gh, name, organization=None, private=False, description='', check_mode=False):
+    result = dict(
+        changed=False,
+        repo=dict())
+    if organization:
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+
+    repo = None
+    try:
+        repo = target.get_repo(name=name)
+        result['repo'] = repo.raw_data
+    except UnknownObjectException:
+        if not check_mode:
+            repo = target.create_repo(
+                name=name, private=private, description=description)
+            result['repo'] = repo.raw_data
+
+        result['changed'] = True
+
+    changes = {}
+    if repo is None or repo.raw_data['private'] != private:
+        changes['private'] = private
+    if repo is None or repo.raw_data['description'] != description:
+        changes['description'] = description
+
+    if changes:
+        if not check_mode:
+            repo.edit(**changes)
+
+        result['repo'].update({
+            'private': repo._private.value if not check_mode else private,
+            'description': repo._description.value if not check_mode else description
```

Comment:
This reduces the diff the next time someone adds something new:
```suggestion
            'description': repo._description.value if not check_mode else description,
```